### PR TITLE
 arrow: Update to 12.0.0

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Comment by raulcd** [comment]
CI seems to be failing due to `check_option` not found:
```
 mingw-w64-arrow/PKGBUILD: line 58: check_option: command not found

[MSYS2 CI] Processing changes

PKGBUILD: line 58: check_option: command not found
./.ci/ci-build.sh: line 113: test: too many arguments
```
This seems to be added here: https://github.com/msys2/MINGW-packages/commit/2930e301700d3d0c728c89c92fb9fa5320618ba4
@MehdiChinoune any idea what might be the issue? Sorry for pinging you directly but I see you were the one that added those changes

**Comment by Biswa96** [comment]
It seems that CI failed due to SHA mismatch. Please recheck https://github.com/msys2/MINGW-packages/pull/17050/checks#step:11:437

**Comment by Biswa96** [comment]
The link in the PKGBUILD is probably invalid. It downloads two text files with `unknown action [download\]` string.

**Comment by raulcd** [comment]
I am unsure why `gdal` is failing on the `UCRT64` job:
```
  [584/1236] Building CXX object frmts/hdf5/CMakeFiles/gdal_HDF5.dir/hdf5dataset.cpp.obj
  FAILED: frmts/hdf5/CMakeFiles/gdal_HDF5.dir/hdf5dataset.cpp.obj 
  D:\M\msys64\ucrt64\bin\g++.exe -DDONT_DEPRECATE_SPRINTF -DENABLE_HDF5_GLOBAL_LOCK -DGDAL_COMPILATION -DH5_BUILT_AS_DYNAMIC_LIB -DH5_BUILT_AS_STATIC_LIB -DHAVE_AVX_AT_COMPILE_TIME -DHAVE_SSE_AT_COMPILE_TIME -DHAVE_SSSE3_AT_COMPILE_TIME -DUSE_IMPORT_EXPORT -DWIN32 -IC:/_/B/src/gdal-3.6.4/apps -IC:/_/B/src/gdal-3.6.4/alg -IC:/_/B/src/gdal-3.6.4/gcore -IC:/_/B/src/build-UCRT64-shared/gcore -IC:/_/B/src/gdal-3.6.4/port -IC:/_/B/src/build-UCRT64-shared/port -IC:/_/B/src/gdal-3.6.4/ogr -IC:/_/B/src/gdal-3.6.4/ogr/ogrsf_frmts -IC:/_/B/src/gdal-3.6.4/frmts -isystem D:/M/msys64/ucrt64/include -march=nocona -msahf -mtune=generic -O2 -pipe -fvisibility=hidden -O3 -DNDEBUG -Wall -Wextra -Winit-self -Wunused-parameter -Wmissing-declarations -Wlogical-op -Wshadow -Wmissing-include-dirs -Wformat -Werror=format-security -Wno-format-nonliteral -Werror=vla -Wno-clobbered -Wdate-time -Wnull-dereference -Wextra-semi -Wfloat-conversion -Wnon-virtual-dtor -Woverloaded-virtual -Wsuggest-override -fno-operator-names -Wzero-as-null-pointer-constant -MD -MT frmts/hdf5/CMakeFiles/gdal_HDF5.dir/hdf5dataset.cpp.obj -MF frmts\hdf5\CMakeFiles\gdal_HDF5.dir\hdf5dataset.cpp.obj.d -o frmts/hdf5/CMakeFiles/gdal_HDF5.dir/hdf5dataset.cpp.obj -c C:/_/B/src/gdal-3.6.4/frmts/hdf5/hdf5dataset.cpp
  In file included from D:/M/msys64/ucrt64/include/c++/13.1.0/ext/string_conversions.h:43,
                   from D:/M/msys64/ucrt64/include/c++/13.1.0/bits/basic_string.h:4108,
                   from D:/M/msys64/ucrt64/include/c++/13.1.0/string:54,
                   from C:/_/B/src/gdal-3.6.4/port/cpl_error.h:245,
                   from C:/_/B/src/gdal-3.6.4/gcore/gdal.h:46,
                   from C:/_/B/src/gdal-3.6.4/gcore/gdal_priv.h:61,
                   from C:/_/B/src/gdal-3.6.4/gcore/gdal_pam.h:36,
                   from C:/_/B/src/gdal-3.6.4/frmts/hdf5/hdf5dataset.h:37,
                   from C:/_/B/src/gdal-3.6.4/frmts/hdf5/hdf5dataset.cpp:35:
  D:/M/msys64/ucrt64/include/c++/13.1.0/cstdlib:79:15: fatal error: stdlib.h: No such file or directory
     79 | #include_next <stdlib.h>
        |               ^~~~~~~~~~
  compilation terminated.
```
@kou any idea?

**Comment by mmuetzel** [comment]
That's an error that intermittently happens for UCRT64 for still unknown reasons afaict.
It usually builds just fine the next time...

File: mingw-w64-arrow/PKGBUILD [hunk 1]
```diff
@@ -38,9 +38,9 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-clang"
              "${MINGW_PACKAGE_PREFIX}-gtk-doc"
              "${MINGW_PACKAGE_PREFIX}-rapidjson")
-source=("apache-arrow-${pkgver}.tar.gz"::"https://www.apache.org/dyn/closer.lua?action=download\&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz"
-        "apache-arrow-${pkgver}.tar.gz.asc"::"https://www.apache.org/dyn/closer.lua?action=download\&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz.asc")
-sha256sums=('2dd8f0ea0848a58785628ee3a57675548d509e17213a2f5d72b0d900b43f5430'
+source=("apache-arrow-${pkgver}.tar.gz"::"https://dlcdn.apache.org/arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz"
+        "apache-arrow-${pkgver}.tar.gz.asc"::"https://dlcdn.apache.org/arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz.asc")
```
Comment (kou) [inline mingw-w64-arrow/PKGBUILD:42]: Can we keep `https://www.apache.org/dyn/closer.lua?...` URL?
It's the official URL recommended by ASF.

See also: https://infra.apache.org/release-download-pages.html#download-scripts

Comment (raulcd) [reply mingw-w64-arrow/PKGBUILD]: Thanks @kou I didn't know about this policy. Removing the back-slash from the original URL seems to have fixed the sha failure. Not sure why that worked previously :) but I've reverted back to the original URL

```
  ==> Validating source files with sha256sums...
      apache-arrow-12.0.0.tar.gz ... FAILED
```

**Comment by raulcd** [comment]
:tada: CI is green :)

**Pull request merged**